artemis-core-client/src/main/java/org/apache/activemq/artemis/spi/core/remoting/ssl/OpenSSLContextFactoryProvider.java

```diff
@@ -25,7 +25,7 @@
    private static final OpenSSLContextFactory FACTORY;
    static {
       OpenSSLContextFactory factoryWithHighestPrio = null;
-      for (OpenSSLContextFactory factory : ServiceLoader.load(OpenSSLContextFactory.class)) {
```

Comment:
Can't you use ThreadContextClass Loader on all the tests here?

it would be nice to have a test.

Reply:
OSGi in certain conditions has issues with ServiceLoader.load(<class>.class) which uses Thread.currentThread().getContextClassLoader() because that classloader is not always the same as the one that is available to the bundle throughout its life. This change should be transparent for any environment that doesn’t juggle class-loaders like OSGi since either way would end up returning the same class-loader.

As far as writing a test its difficult since the conditions i was seeing errors are hard to recreate see the JIRA for details.

Reply:
OSGi in certain conditions has issues with ServiceLoader.load(<class>.class) which uses Thread.currentThread().getContextClassLoader() because that classloader is not always the same as the one that is available to the bundle throughout its life. This change should be transparent for any environment that doesn’t juggle class-loaders like OSGi since either way would end up returning the same class-loader.

 As far as writing a test its difficult since the conditions i was seeing errors are hard to recreate see the JIRA for details.